python/dolfinx/fem/assemble.py

```diff
@@ -245,6 +246,9 @@ def assemble_matrix(
     Note:
         The returned matrix is not finalised, i.e. ghost values are not
         accumulated.
+
+    Note:
+        Diagonal entries are only inserted for matrices with the same trial and test space.
```

Comment:
Notes are more asides. As this is a critical point I would add it to the argument as 'for constrained degrees of freedom belonging to the same trial and test space.'